src/MemoryTracing/MemoryInfoProducer.cpp

```diff
@@ -63,7 +63,7 @@ void SystemMemoryInfoProducerRunFn(MemoryInfoListener* listener, int32_t /*pid*/
 
 std::unique_ptr<MemoryInfoProducer> CreateSystemMemoryInfoProducer(MemoryInfoListener* listener,
                                                                    uint64_t sampling_period_ns,
-                                                                   int32_t pid) {
+                                                                   uint32_t pid) {
```

Comment:
this also should be int32_t since this is linux-specific component.

Also every time in linux code we use uint32_t as pid we should CHECK that it is <= MAX_INT

Reply:
Yes I agree, done. Also going through all the similar occurences and reverting to int32_t where possible.